File: PROPMANS.ACCOUNT_SEARCH/AccountLookupForm.cs
using System;
using System.Collections.Generic;
using System.Windows.Forms;


namespace PROPMANS.ACCOUNT_SEARCH
{
    public enum AccountLookupDisplay
    {
        ActiveUnitOwnerOnly,
        CUstomerOnly,
        AllUnits
    }
    public partial class AccountLookupForm : ComponentFactory.Krypton.Toolkit.KryptonForm , IAccountLookUp
    {
        private AccountLookUpPresenter presenter;

        public BCL.Accounts.Customer SelectedCustomer
        {
            get { return (this.CustomerAccountsGridView.SelectedRows[0].DataBoundItem as CustomerDisplay).Customer; }
        }
        public BCL.Accounts.CustomerAccount SelectedAccount
        {
            get { return (this.CustomerAccountsGridView.SelectedRows[0].DataBoundItem as CustomerAccountDisplay).CustAccount; }
        }

        public BCL.Accounts.UnitInventory SelectedUnit
        {
            get { return (this.CustomerAccountsGridView.SelectedRows[0].DataBoundItem as AllUnitsDisplay).Unit; }
        }

        public bool DisplayInactiveCheckBox
        {
            set { DisplayInactiveAccountsCheckBox.Visible = value; }
        }
        public bool DisplayNewButton
        {
            set { btnNewCustomer.Visible = value; }
        }
        public bool DisplayUnitNumberTextbox
        {
            set { ByUnitNumberRadioButton.Enabled = value; }
        }

        public SearchAccountCriteria SetSearchCriteria
        {
            set
            {
                switch (value)
                {
                    case SearchAccountCriteria.UnitNumber:
                        ByUnitNumberRadioButton.Checked = true;
                        break;
                    case SearchAccountCriteria.LastName:
                        ByLastNameRadioButton.Checked = true;
                        break;
                    case SearchAccountCriteria.FirstName:
                        ByFirstNameRadioButton.Checked = true;
                        break;
                    default:
                        break;
                }
            }
        }

        public SearchAccountCriteria SearchByOption
        {
            get
            {
                SearchAccountCriteria criteria = SearchAccountCriteria.UnitNumber;
                if (ByUnitNumberRadioButton.Checked)
                {
                    criteria = SearchAccountCriteria.UnitNumber;
                }
                if (ByLastNameRadioButton.Checked)
                {
                    criteria = SearchAccountCriteria.LastName;
                }
                if (ByFirstNameRadioButton.Checked)
                {
                    criteria = SearchAccountCriteria.FirstName;
                }

                return criteria;
            }

        }
       
        public string SearchValue
        {
            get
            {
                return SearchValueTextBox.Text.Trim();
            }
        }
       
        public bool DisplayInactiveAccounts
        {
            get { return DisplayInactiveAccountsCheckBox.Checked; }
        }

        public string StatusChange
        {
            set { ResultCountLabel.Text = value; }
        }


        public void LoadUnits(List<AllUnitsDisplay> units)
        {
            btnSelect.Enabled = units.Count > 0;
            this.CustomerAccountsGridView.DataSource = units;
        }

        public void LoadCustomers(List<CustomerDisplay> customers)
        {
            btnSelect.Enabled = customers.Count > 0;
            this.CustomerAccountsGridView.DataSource = customers;
        }
        
        public void LoadCustomerAccounts(List<CustomerAccountDisplay> accounts)
        {
            btnSelect.Enabled = accounts.Count > 0;
            this.CustomerAccountsGridView.DataSource = accounts;
        }

        public AccountLookupForm(AccountLookupDisplay displayType)
        {
            InitializeComponent();

            if (displayType == AccountLookupDisplay.ActiveUnitOwnerOnly)
            {
                presenter = new UnitOwnerOnlyAccountLookupPresenter(this);
            }
            else if (displayType == AccountLookupDisplay.AllUnits) {
                presenter = new AllUnitsAccountLookupPresenter(this);
            }
            else
            {
                presenter = new CustomerOnlyAccountLookUpPresenter(this);
            }
           
        }

        private void AccountLookupForm_Load_1(object sender, EventArgs e)
        {
            CustomerAccountsGridView.ReadOnly = true;
            btnSelect.Enabled = false;
        }
        
        private void SearchOptionRadioButton_CheckedChanged(object sender, EventArgs e)
        {
            HighlightField();
        }

        private void btnSearch_Click(object sender, EventArgs e)
        {
            if (string.IsNullOrWhiteSpace(SearchValueTextBox.Text))
            {
                HighlightField();
            }
            else
            {
                presenter.Search();
            }
        }

        private void txtViewAll_Click(object sender, EventArgs e)
        {
            SearchValueTextBox.Text = string.Empty;
            HighlightField();
            presenter.ViewAll();
        }
       
        private void HighlightField(){
            SearchValueTextBox.SelectAll();
            SearchValueTextBox.Focus();
        }

        private void ClearSearchValueButton_Click(object sender, EventArgs e)
        {
            SearchValueTextBox.Text = string.Empty;
            HighlightField();
        }

        private void btnSelect_Click(object sender, EventArgs e)
        {
            this.DialogResult = DialogResult.OK;
            this.Close();
        }

        private void btnCancel_Click(object sender, EventArgs e)
        {
            this.DialogResult = DialogResult.Cancel;
            this.Close();
        }

        private void btnRefresh_Click(object sender, EventArgs e)
        {
            presenter.RefreshLookup();
        }

        private void btnNewCustomer_Click(object sender, EventArgs e)
        {
            if (MessageBox.Show("Have you search the name of the tenant before we proceed?","New Tenant Confirmation", MessageBoxButtons.YesNo, MessageBoxIcon.Question) == System.Windows.Forms.DialogResult.Yes)
            {
                using (NewTenantInfoForm frm = new NewTenantInfoForm())
                {
                    if (frm.ShowDialog() == System.Windows.Forms.DialogResult.OK)
                    {
                        presenter.RefreshLookup();
                        this.SetSearchCriteria = SearchAccountCriteria.LastName;
                        this.SearchValueTextBox.Text = frm.CustomerLastName;
                        presenter.Search();
                    }
                }
            }
            else
            {
                SearchValueTextBox.Focus();
            }
           
        }

        private void SearchValueTextBox_TextChanged(object sender, EventArgs e)
        {
            if (string.IsNullOrWhiteSpace(SearchValueTextBox.Text))
            {
                this.AcceptButton = btnViewAll;
            }
            else
            {

                this.AcceptButton = btnSearch;
              
            }
        }
    }
}

File: PROPMANS.BASE/Account Lookup/AccountLookupForm.vb
'***********************************************************************
' Assembly         : PROPMANS_UI
' Author           : Mark Angelo Rivo
' Created          : 04-16-2012
'
' Last Modified By : 
' Last Modified On : 
' Description      : 
'
' Copyright        : (c) Phinma Properties. All rights reserved.
'***********************************************************************
Imports System.Windows.Forms
Imports ComponentFactory.Krypton.Toolkit
Imports BCL.Common


Public Class AccountLookupForm

    Private gridDisplay As GridDisplayType
    Private customerTypeSearch As CustomerType
    Private displayInactiveCheckBox As Boolean = False

    Private _bloaded As Boolean

    Private _accountID As String
    Public Property AccountID() As String
        Get
            Return grid.CurrentRow.Cells("account_id").Value.ToString
        End Get
        Private Set(value As String)
            _accountID = value
        End Set
    End Property

    Private _customerID As String
    Public Property CustomerID() As String
        Get
            Return grid.CurrentRow.Cells("cust_id").Value.ToString
        End Get
        Private Set(value As String)
            _customerID = value
        End Set
    End Property

    Public ReadOnly Property LeaseID() As String
        Get
            Return grid.CurrentRow.Cells("lease_id").Value.ToString
        End Get
    End Property


    Public ReadOnly Property UnitNumber() As String
        Get
            Return grid.CurrentRow.Cells("unit_number").Value.ToString
        End Get
    End Property

    Public ReadOnly Property UnitClass() As FeeUnitClass
        Get
            Try
                If grid.CurrentRow.Cells("unit_class").Value.ToString = "PARKING SLOT" Then
                    Return FeeUnitClass.ParkingUnit
                Else
                    Return FeeUnitClass.CondoUnit
                End If
            Catch ex As Exception
                Return FeeUnitClass.CondoUnit
            End Try

        End Get
    End Property

    Public ReadOnly Property CustomerType As String
        Get
            Return grid.CurrentRow.Cells("cust_type").Value.ToString
        End Get
    End Property

    Public ReadOnly Property CustomerFirstName() As String
        Get
            Return grid.CurrentRow.Cells("first_name").Value.ToString
        End Get
    End Property
    Public ReadOnly Property CustomerMiddleName() As String
        Get
            Return grid.CurrentRow.Cells("middle_name").Value.ToString
        End Get
    End Property
    Public ReadOnly Property CustomerLastName() As String
        Get
            Return grid.CurrentRow.Cells("last_name").Value.ToString
        End Get
    End Property
    Public ReadOnly Property CustomerFullName() As String
        Get
            Return String.Format("{0}, {1} {2}", CustomerLastName, CustomerFirstName, CustomerMiddleName)
        End Get
    End Property

    Private SearchType As SearchAccountType

    Public Sub New(Optional displayType As GridDisplayType = GridDisplayType.Full,
                   Optional custType As CustomerType = BCL.Common.CustomerType.Both,
                   Optional displayInactiveChkBox As Boolean = True)
        Me.gridDisplay = displayType
        Me.customerTypeSearch = custType
        Me.displayInactiveCheckBox = displayInactiveChkBox
        ' This call is required by the designer.
        InitializeComponent()

        ' Add any initialization after the InitializeComponent() call.

    End Sub
    Private Sub CustomerLookupForm_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Load
        SetUIDefaults()
    End Sub

    Private Sub btnSearch_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnSearch.Click
        btnSelect.Visible = False
        If Common.HasValue(SearchValueTextBox) Then
            Dim searchvalue As String = SearchValueTextBox.Text.Trim
            _bloaded = False

            If Me.gridDisplay = GridDisplayType.Full Then
                grid.DataSource = GlobalReference.CustomerAccount.LoadAccounts(SearchType, searchvalue, Me.customerTypeSearch, DisplayInactiveAccountsCheckBox.Checked)
            Else
                grid.DataSource = GlobalReference.CustomerAccount.LoadCustomers(SearchType, searchvalue)
            End If

            _bloaded = True

            If grid.RowCount > 0 Then
                ResultCountLabel.Text = grid.RowCount & " RECORD(S) FOUND"
                btnSelect.Visible = True
            Else
                SearchValueTextBox.SelectAll()
                SearchValueTextBox.Focus()
                ResultCountLabel.Text = "NO RECORD(S) FOUND"
            End If
        Else
            If Me.gridDisplay = GridDisplayType.Full Then
                grid.DataSource = GlobalReference.CustomerAccount.LoadAccounts(Me.customerTypeSearch, DisplayInactiveAccountsCheckBox.Checked)
            Else
                grid.DataSource = GlobalReference.CustomerAccount.LoadCustomers()
            End If

            btnSelect.Visible = True
            ResultCountLabel.Text = grid.RowCount & " RECORD(S) FOUND"
            SearchValueTextBox.Focus()
        End If

    End Sub


    Private Sub btnSelect_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnSelect.Click
        Me.DialogResult = Windows.Forms.DialogResult.OK
        Me.Close()
    End Sub

    Private Sub btnCancel_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnCancel.Click
        Me.DialogResult = Windows.Forms.DialogResult.Cancel
        Me.Close()
    End Sub

    Private Sub btnReset_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles ClearSearchValueButton.Click
        SearchValueTextBox.Text = String.Empty
        SearchValueTextBox.Focus()
    End Sub

    Private Sub btnRefresh_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnRefresh.Click
        GlobalReference.CustomerAccount.Refresh()
        btnSearch_Click(Nothing, Nothing)
    End Sub


    Private Sub Grid_CellFormatting(ByVal sender As Object, ByVal e As System.Windows.Forms.DataGridViewCellFormattingEventArgs)
        If grid.Columns(e.ColumnIndex).Name = "customername" Then
            ' e.FormattingApplied
            Dim row As DataGridViewRow = grid.Rows(e.RowIndex)
            e.Value = String.Format("{0}, {1} {2}", row.Cells("last_name").Value, row.Cells("first_name").Value, row.Cells("middle_name").Value)
        End If
    End Sub

    Private Sub SetUIDefaults()
        Me.CancelButton = btnCancel
        Me.AcceptButton = btnSearch


        If Me.gridDisplay = GridDisplayType.Full Then
            GridHelper.SetGridDisplay(grid, New AccountLookupGridDisplay)
            ByUnitNumberRadioButton.Checked = True
        Else
            GridHelper.SetGridDisplay(grid, New CustomerOnlyLookupGridDisplay)
            Me.ByUnitNumberRadioButton.Enabled = False
            Me.ByLastNameRadioButton.Checked = True
        End If

        Me.DisplayInactiveAccountsCheckBox.Visible = IIf(displayInactiveCheckBox = False, False, (Me.gridDisplay = GridDisplayType.Full))


        AddHandler grid.CellFormatting, AddressOf Grid_CellFormatting

        btnSelect.Visible = False
        'btnRefresh.Visible = False


        SearchValueTextBox.Focus()
    End Sub


    Private Sub RadioButton_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles ByUnitNumberRadioButton.CheckedChanged, ByLastNameRadioButton.CheckedChanged, ByFirstNameRadioButton.CheckedChanged

        Select Case DirectCast(sender, KryptonRadioButton).Tag
            Case "ByUnitNumber" ' search by Unit No
                SearchType = SearchAccountType.UnitNumber
            Case "ByLastName" ' search by Last Name
                SearchType = SearchAccountType.LastName
            Case "ByFirstName" ' search by First Name
                SearchType = SearchAccountType.Firstname
        End Select

        SearchValueTextBox.SelectAll()
        SearchValueTextBox.Focus()
    End Sub


End Class